_posts/2023-04-02-writing-custom-pod-security-providers.md

```diff
@@ -0,0 +1,269 @@
+  In this example, we will ignore it for simplicity.
+
+For our example with the read-only root filesystem, we do not care about the Pod Security Context because our configuration is part of the container Security Context.
+So we set all the Pod methods to simply return `null` which means that no Pod Security Context should be set.
+For example:
+
+```java
+    @Override
+    public PodSecurityContext kafkaPodSecurityContext(PodSecurityProviderContext context) {
+        return null;
+    }
+```
+
+And in the container methods, we always return the Security Context which enables the read-only root filesystem:
+
+```java
+    @Override
+    public SecurityContext kafkaContainerSecurityContext(ContainerSecurityProviderContext context) {
+        return new SecurityContextBuilder()
+                .withReadOnlyRootFilesystem()
+                .build();
+    }
+
+    @Override
+    public SecurityContext kafkaInitContainerSecurityContext(ContainerSecurityProviderContext context) {
+        return new SecurityContextBuilder()
+                .withReadOnlyRootFilesystem()
+                .build();
+    }
+```
+
+You can find the full source code in the GitHub repository linked at the end of the blog post.
+
+#### Extending an existing policy
+
+If you want to do only some small changes to an existing provider, you do not need to implement everything from scratch.
+You can just extend the provider and re-implement only the methods you want to change.
+Strimzi currently includes two providers: [`BaselinePodSecurityProvider` and `RestrictedPodSecurityProvider`](https://github.com/strimzi/strimzi-kafka-operator/tree/main/api/src/main/java/io/strimzi/plugin/security/profiles/impl)
+
+For our example, we are going to take the [`RestrictedPodSecurityProvider`](https://github.com/strimzi/strimzi-kafka-operator/blob/main/api/src/main/java/io/strimzi/plugin/security/profiles/impl/RestrictedPodSecurityProvider.java).
+It configures the security context of the Strimzi operands to match Kubernetes' _restricted_ security profile.
+When you try to use the Kafka Connect Build and its [Kaniko](https://github.com/GoogleContainerTools/kaniko) builder with this provider, it will [throw an exception](https://github.com/strimzi/strimzi-kafka-operator/blob/main/api/src/main/java/io/strimzi/plugin/security/profiles/impl/RestrictedPodSecurityProvider.java#L94-L103) because the Kaniko container builder does not work under the _restricted_ profile.
+Imagine, you want to use the `RestrictedPodSecurityProvider` to harden the operand Pods.
+But you want to use the Kafka Connect Build feature at the same time without any restrictions.
+
+In such case, you can simply extend the `RestrictedPodSecurityProvider` and override the `kafkaConnectBuildContainerSecurityContext` method with your implementation which will just let it run instead of throwing an exception:
```

Comment:
```suggestion
In such a case, you can simply extend the `RestrictedPodSecurityProvider` and override the `kafkaConnectBuildContainerSecurityContext` method with your implementation, which will just let it run instead of throwing an exception:
```